libr/anal/xrefs.c

```diff
@@ -29,12 +29,23 @@ static void XREFKEY(char * const key, const size_t key_len,
 }
 
 R_API int r_anal_xrefs_load(RAnal *anal, const char *prjfile) {
-	char *path, *db = r_str_newf (R2_HOMEDIR"/projects/%s.d", prjfile);
+		char *path, *db;
 	ut8 found = 0;
 	SdbListIter *it;
 	SdbNs *ns;
-	if (!db) return false;
-	path = r_str_home (db);
+
+	if(!prjfile || strlen(prjfile) < 1) return false;
```

Comment:
if strlen ()<1 it means strlen == 0, therefor the only thing you need to check here if the first char is not \0, strlen is an unnecessary slowdown